packages/react-sdk/src/components/CallControls/ScreenShareButton.tsx

```diff
@@ -26,10 +26,11 @@ export const ScreenShareButton = (props: ScreenShareButtonProps) => {
   const callSettings = useCallSettings();
   const isScreenSharingAllowed = callSettings?.screensharing.enabled;
 
-  const { screenShare, isMute: amIScreenSharing } = useScreenShareState();
-  const disableScreenShareButton = amIScreenSharing
-    ? isSomeoneScreenSharing || isScreenSharingAllowed === false
-    : false;
+  const { screenShare, optimisticIsMute } = useScreenShareState();
+  const amIScreenSharing = !optimisticIsMute;
+  const disableScreenShareButton =
+    !amIScreenSharing &&
+    (isSomeoneScreenSharing || isScreenSharingAllowed === false);
```

Comment:
This was breaking my brain until I realized that the `amIScreenSharing` variable was named incorrectly.